File: automation-test-selenium/src/main/java/org/suntower/pages/auth/RegisterPage.java
package org.suntower.pages.auth;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.suntower.core.BasePage;

public class RegisterPage extends BasePage {
  private final By emailInput = anyCss("[data-testid='register-email']", "input[name='email']", "input[type='email']");
  private final By submitButton = anyCss("[data-testid='register-submit']", "button[type='submit']");

  public RegisterPage(WebDriver driver) {
    super(driver);
  }

  public void open() {
    visit("/register");
  }

  public void waitForLoaded() {
    waitForUrlContains("/register");
    waitForVisible(emailInput);
    waitForVisible(submitButton);
  }

  public void requestRegistrationCode(String email) {
    fill(emailInput, email);
    click(submitButton);
  }

  public String emailValue() {
    return value(emailInput);
  }
}

File: automation-test-playwright/pages/auth/RegisterPage.ts
import { type Locator, type Page } from "@playwright/test";
import { BasePage } from "../core/BasePage";

export class RegisterPage extends BasePage {
  readonly emailInput: Locator;
  readonly submitButton: Locator;

  /**
   * Initializes this page object.
   */
  constructor(page: Page) {
    super(page);
    this.emailInput = this.firstVisible(
      this.anyLocator('[data-testid="register-email"]', 'input[name="email"]', 'input[type="email"]')
    );
    this.submitButton = this.firstVisible(this.anyLocator('[data-testid="register-submit"]', 'button[type="submit"]'));
  }

  /**
   * Opens the page.
   */
  async open(): Promise<void> {
    await this.visit("/register");
  }

  async waitForLoaded(): Promise<void> {
    await this.page.waitForURL(/\/register/);
    await this.waitForVisible(this.emailInput);
    await this.waitForVisible(this.submitButton);
  }

  async requestRegistrationCode(email: string): Promise<void> {
    await this.emailInput.fill(email);
    await this.submitButton.click();
  }
}
